src/libasr/codegen/asr_to_llvm.cpp

```diff
@@ -7981,9 +7981,7 @@ class ASRToLLVMVisitor : public ASR::BaseVisitor<ASRToLLVMVisitor>
         for (size_t i=0; i<x.n_values; i++) {
             if (i != 0) {
                 fmt.push_back("%s");
-                if (global_sep_space &&
-                    !(ASRUtils::is_character(*ASRUtils::expr_type(x.m_values[i]))
-                        && ASRUtils::is_character(*ASRUtils::expr_type(x.m_values[i - 1])))) {
+                if (global_sep_space) {
```

Comment:
The way to fix this such that we print spaces when two consecutive arguments are strings in LPython and not print spaces for the same in LFortran, is to support this in AST->ASR Translation.

Reply:
You will need to figure out how to update the middle-ends of both lfortran and lpython such that both of them work as expected when two consecutive arguments are strings.

Reply:
@Shaikh-Ubaid thanks for this! I will try to create a solution that works for both. :+1:

Reply:
See if your solution works for both lpython and lfortran (test it locally on your system or send a PR to lfortran so it gets tested at the CI). If no `run` reference tests change for lfortran, then it works.

You can also test your current approach with lfortran. If it works currently, then I think its good.